esrally/driver/driver.py

```diff
@@ -313,42 +336,157 @@ def load_local_config(coordinator_config):
+    def receiveMsg_PrepareTrack(self, msg, sender):
+        self.original_sender = sender
+        # load node-specific config to have correct paths available
+        self.cfg = load_local_config(msg.config)
+        self.data_root_dir = self.cfg.opts("benchmarks", "local.dataset.cache")
+        tpr = TrackProcessorRegistry(self.cfg)
+        self.track = msg.track
+
+        self.logger.info("Preparing track [%s]", self.track.name)
+        if self.track.has_plugins:
+            self.logger.info("Reloading track [%s] to ensure plugins are up-to-date.", self.track.name)
+            # the track might have been loaded on a different machine (the coordinator machine) so we force a track
+            # update to ensure we use the latest version of plugins.
+            load_track_plugins(self.cfg, self.track.name, register_track_processor=tpr.register_track_processor,
+                               force_update=True)
+
+        if tpr.processors():
+            # do this locally...
+            for processor in tpr.processors():
+                processor.on_after_load_track(self.track)
+            # ...and delegate the rest
+            self.children = [self._create_task_worker() for _ in range(num_cores(self.cfg))]
+            for processor in tpr.processors():
+                self.processors.put(processor)
+            self._seed_tasks(self.processors.get())
+            self.send_to_children_and_transition(self, StartTaskLoop(self.track.name, self.cfg), self.Status.INITIALIZING,
+                                                 self.Status.PROCESSOR_RUNNING)
+        else:
+            self.send(sender, TrackPrepared())
+
+    def resume(self):
+        if not self.processors.empty():
+            self._seed_tasks(self.processors.get())
+            self.send_to_children_and_transition(self, StartTaskLoop(self.track.name, self.cfg), self.Status.PROCESSOR_COMPLETE,
+                                                 self.Status.PROCESSOR_RUNNING)
+        else:
+            self.send(self.original_sender, TrackPrepared())
+
+    def _seed_tasks(self, processor):
+        self.tasks = list(processor.on_prepare_track(self.track, self.data_root_dir))
+
+    def _create_task_worker(self):
+        child = self.createActor(TaskExecutionActor)
+        return child
+
+    @actor.no_retry("track preparator")  # pylint: disable=no-value-for-parameter
+    def receiveMsg_ReadyForWork(self, msg, sender):
+        if self.tasks:
+            next_task = self.tasks.pop()
+        else:
+            next_task = None
+        new_msg = DoTask(next_task, self.cfg)
+        self.logger.debug("Track Preparator sending %s to %s", vars(new_msg), sender)
+        self.send(sender, new_msg)
+
+    def receiveMsg_WorkerIdle(self, msg, sender):
+        self.transition_when_all_children_responded(sender, msg, self.Status.PROCESSOR_RUNNING,
+                                                    self.Status.PROCESSOR_COMPLETE, self.resume)
+
+
+def num_cores(conf):
```

Comment:
Nit: So far we've referred to the configuration either as `config` or `cfg`, maybe we should use on of these here as well.